src/shared/use-data-value-set/use-data-value-set.js

```diff
@@ -11,38 +11,18 @@ function mapDataValuesToFormInitialValues(dataValues) {
         return {}
     }
 
-    const formInitialValues = dataValues.reduce(
-        (
-            acc,
-            {
-                dataElement,
-                categoryOptionCombo,
-                value,
-                comment,
-                storedBy,
-                followUp,
-                lastUpdated,
-            }
-        ) => {
-            const dataValueData = {
-                value,
-                dataElement,
-                comment,
-                storedBy,
-                followUp,
-                lastUpdated,
-            }
-
-            if (!acc[dataElement]) {
-                acc[dataElement] = { [categoryOptionCombo]: dataValueData }
-            } else {
-                acc[dataElement][categoryOptionCombo] = dataValueData
+    const formInitialValues = dataValues.reduce((acc, dataValueData) => {
+        if (!acc[dataValueData.dataElement]) {
+            acc[dataValueData.dataElement] = {
+                [dataValueData.categoryOptionCombo]: dataValueData,
```

Comment:
I didn't really understand why some of the fields were filtered out. I think it's much clearer when this contains the exact data we have in the cache/get from the API.

Reply:
Agreed.